R/projpred.R

```diff
@@ -122,53 +116,38 @@ get_refmodel.brmsfit <- function(object, newdata = NULL, resp = NULL,
     .extract_model_data(object, newdata = newdata, resp = resp, ...)
   }
   
-  # The default `ref_predfun` from projpred does not set `allow_new_levels`, so
-  # use a customized `ref_predfun`:
-  ref_predfun <- function(fit, newdata = NULL) {
-    # Setting a seed is necessary for reproducible sampling of group-level
-    # effects for new levels:
-    if (exists(".Random.seed", envir = .GlobalEnv)) {
-      rng_state_old <- get(".Random.seed", envir = .GlobalEnv)
-      on.exit(assign(".Random.seed", rng_state_old, envir = .GlobalEnv))
+  if (!is_ordinal(family)) {
+    # The default `ref_predfun` from projpred does not set `allow_new_levels`,
+    # so use a customized `ref_predfun`:
+    ref_predfun <- function(fit, newdata = NULL) {
+      # Setting a seed is necessary for reproducible sampling of group-level
+      # effects for new levels:
+      if (exists(".Random.seed", envir = .GlobalEnv)) {
+        rng_state_old <- get(".Random.seed", envir = .GlobalEnv)
+        on.exit(assign(".Random.seed", rng_state_old, envir = .GlobalEnv))
+      }
+      set.seed(refprd_seed)
+      t(posterior_linpred(fit,
+                          newdata = newdata,
+                          allow_new_levels = TRUE,
+                          sample_new_levels = "gaussian"))
     }
-    set.seed(refprd_seed)
-    t(posterior_linpred(fit,
+  } else {
+    # Also use argument `incl_thres` of posterior_linpred():
+    ref_predfun <- function(fit, newdata = NULL) {
+      # Setting a seed is necessary for reproducible sampling of group-level
+      # effects for new levels:
+      if (exists(".Random.seed", envir = .GlobalEnv)) {
+        rng_state_old <- get(".Random.seed", envir = .GlobalEnv)
+        on.exit(assign(".Random.seed", rng_state_old, envir = .GlobalEnv))
+      }
+      set.seed(refprd_seed)
+      posterior_linpred(fit,
```

Comment:
here, you don't transpose via `t()`. Is this intentional?

Reply:
Yes, I decided to only accept S x N x C_lat arrays (S: number of draws, N: number of observations, C_lat: number of latent response categories, i.e., on linear predictor scale) as output of `ref_predfun` in case of the augmented-data projection. That avoids an unnecessary permutation of the array margins.

Reply:
I documented this in **projpred**, so I users should notice this.

Reply:
Thanks!